guides/20240823_how_to_setup_postgresql_playground_in_daytona.md

```diff
@@ -0,0 +1,372 @@
+
+## Overview of Daytona
+
+Daytona is a self hosted and secure open source development environment manager that uses configurations from a project's repository to build workspace and provision the workspace in a platform of your choice. It's innovative and incredibly easy for all levels including beginners to get started with it.
+
+Daytona provides some interesting features which make it one the best products in the area of simplifying development environments for both enterprise and individual levels.
+
+Some of those features it boast of include:
+- It has support for popular IDE like VS Code and JetBrains
+- It connect with repository providers like GitHub, GitLab, BitBucket and Gitea
+- It's very secure. It uses VPN connection to make that possible
+- It has reverse proxy support
+
+For more info about Daytona and it's features, check [here](https://daytona.io)
+
+## Creating a DevContainer for PostgreSQL
+
+A development container (or a devcontainer for short) allows you to use a docker container as a full-featured development environment. It can be configured to meet your development environment needs. It could include tools and runtimes like npm, git, maven, Golang compiler and others too.
+
+For this guide we're going to create a devcontainer for PostgreSQL using a config file. The file is always named `devcontainer.json` with code syntax in it following the correct config specifications. Its a norm to keep the file in an hidden directory call `.devcontainer`
+
+Let's get started. I will use the terminal to create mine in my linux PC but the commands should work fine on Mac terminal or the Windows powershell.
+
+- ### **Step 1**: Create a Directory
+
+  Create a directory with any name of your choice and go into it. I use the name `postgresql-playground-in-daytona` and move into the directory.
+
+  ```bash
+  mkdir postgresql-playground-in-daytona && cd postgresql-playground-in-daytona
+  ```
+
+- ### **Step 2**: Create the `.devcontainer` directory
+
+  Create a hidden directory called `.devcontainer` and enter it. This is where our devcontainer config file will be stored.
+
+  ```bash
+  mkdir .devcontainer && cd .devcontainer
+  ```
+
+- ### **Step 3**: Create `devcontainer.json` file
+
+  Now, create a file called `devcontainer.json` and paste the following code into it and then save it.
+
+  ```json
+  {
+      "name": "PostgreSQL Dev Container Playground",
+      "dockerComposeFile": "docker-compose.yml",
+      "workspaceFolder": "/workspaces/${localWorkspaceFolderBasename}",
+      "service": "postgres",
+      "forwardPorts": [5432],
+      "customizations": {
+        "vscode": {
+            "extensions": ["streetsidesoftware.code-spell-checker"]
+        }
+      },
+      "postCreateCommand": "psql -U ${POSTGRES_USER} -c 'SELECT 1;'"
+  }
```

Comment:
there is also a feature for postgress have you checked that? https://containers.dev/features

Reply:
Yes, I took your advise and it's great